llvm/lib/Target/DirectX/DXILLegalizePass.cpp

```diff
@@ -246,6 +246,61 @@ downcastI64toI32InsertExtractElements(Instruction &I,
   }
 }
 
+static void emitMemcpyExpansion(IRBuilder<> &Builder, Value *Dst, Value *Src,
+                                ConstantInt *Length) {
+
+  uint64_t ByteLength = Length->getZExtValue();
+  if (ByteLength == 0)
```

Comment:
should we error here?  Returning here lets the memcpy  live on past legalization. Feels weird that we would just let a memcpy live on if its malformed.

Reply:
The memcpy gets removed by the `ToRemove.push_back(CI);` after this function returns, no?

Reply:
But its not valid to just remove it, we should error. Your Destination Ptr would be unintialized.

Reply:
Never mind I suppose if that's what the implementer wants then we should honor it.